openmc/source.py

```diff
@@ -2,13 +2,16 @@
 from enum import Enum
 from numbers import Real
 import warnings
+import typing  # imported separately as py3.8 requires typing.Iterable
+from typing import Optional, Union, Tuple
```

Comment:
The test failure is actually a result of this line -- because we are importing `Union` and the `__init__.py` file does a `from openmc.source import *`, it overwrites the _actual_ `openmc.Union` class. To fix this, you can either 1) remove this import and explicitly write `typing.Union` below or 2) set `__all__` so that only specified variables will get caught from the `*` import.